src/fsharp/vs/service.fs

```diff
@@ -2789,7 +2755,7 @@ type BackgroundCompiler(legacyReferenceResolver, projectCacheSize, keepAssemblyC
             parseCacheLock.AcquireLock (fun ltok -> 
                 parseAndCheckFileInProjectCachePossiblyStale.Clear ltok
                 parseAndCheckFileInProjectCache.Clear ltok
-                parseFileInProjectCache.Clear ltok)
+                parseFileInProjectCache.Clear (AssumeAnyCallerThreadWithoutEvidence()))
```

Comment:
As mentioned above, I think it's better for parseFileInProjectCache to still need a lock token